### PR TITLE
Add deprecation notice and further instructions to QuantumInstance migration guide

### PR DESCRIPTION
**Pull request by ElePT**
This is a proposal to address https://github.com/Qiskit/documentation/issues/1696, which doesn't correspond with my original reply.

While updating to the changes in ISA handling doesn't seem to complicated, I realized that updating the guide to use Primitives V2 seems very strange to me (because that is not how the migration was done at the time). The current guide is a (in my opinion, valid) reflection of what the motivations were with primitives V1, and the reason why it was appreciated was because it reflected well the evolution of the code. I also think that it is commonly understood that migration guides are written at a specific point in time, and may require "chaining" with a further migration guide to reach the current state of the package. So, instead of migrating everything to V2, I propose adding a warning at the beginning of the guide pointing to the V1 to V2 migration guide. 

Once again, this is just a proposal, and I am happy to rethink this.

### CODE REVIEW

File: docs/migration-guides/qiskit-quantum-instance.mdx [hunk 1]
```diff
@@ -12,21 +12,30 @@ These include measurement error mitigation, splitting and combining execution to
...
 | --- | --- |
 | [`qiskit.utils.QuantumInstance.execute`](/api/qiskit/0.46/qiskit.utils.QuantumInstance#execute) | [`qiskit.primitives.Sampler.run`](/api/qiskit/qiskit.primitives.Sampler#run) or [`qiskit.primitives.Estimator.run`](/api/qiskit/qiskit.primitives.Estimator#run) |
-| [`qiskit.utils.QuantumInstance.transpile`](/api/qiskit/0.46/qiskit.utils.QuantumInstance#transpile) | [`qiskit.compiler.transpile`](/api/qiskit/compiler#qiskit.compiler.transpile) |
-| [`qiskit.utils.QuantumInstance.assemble`](/api/qiskit/0.46/qiskit.utils.QuantumInstance#assemble) | [`qiskit.compiler.assemble`](/api/qiskit/compiler#qiskit.compiler.assemble) |
+| [`qiskit.utils.QuantumInstance.transpile`](/api/qiskit/0.46/qiskit.utils.QuantumInstance#transpile) | [`qiskit.compiler.transpile`](/api/qiskit/compiler#qiskit.compiler.transpile) or [`qiskit.transpiler.pass_manager.PassManager.run`](/api/qiskit/qiskit.transpiler.pass_manager.PassManager#run)|
+| [`qiskit.utils.QuantumInstance.assemble`](/api/qiskit/0.46/qiskit.utils.QuantumInstance#assemble) | [`qiskit.compiler.assemble`](/api/qiskit/compiler#qiskit.compiler.assemble) [DEPRECATED: no longer part of the workflow]|
```
Comment (beckykd) [inline docs/migration-guides/qiskit-quantum-instance.mdx:25]:
```suggestion
| [`qiskit.utils.QuantumInstance.assemble`](/api/qiskit/0.46/qiskit.utils.QuantumInstance#assemble) | [`qiskit.compiler.assemble`](/api/qiskit/compiler#qiskit.compiler.assemble) [Deprecated: no longer part of the workflow]|
```

File: docs/migration-guides/qiskit-quantum-instance.mdx [hunk 2]
```diff
@@ -12,21 +12,30 @@ These include measurement error mitigation, splitting and combining execution to
...
 
 The remainder of this guide focused on the [`qiskit.utils.QuantumInstance.execute`](/api/qiskit/0.46/qiskit.utils.QuantumInstance#execute) to
 [`qiskit.primitives`](/api/qiskit/primitives) migration path.
 
+<Admonition type="caution">
+  **Deprecation Notice**
+
+  This guide preceeds the introduction of Primitives V2. It was drafted with the intention to guide users from `QuantumInstance` to
```
Comment (beckykd) [inline docs/migration-guides/qiskit-quantum-instance.mdx:33]:
```suggestion
  This guide precedes the introduction of the V2 primitives interface. It was created to guide users from `QuantumInstance` to 
```

File: docs/migration-guides/qiskit-quantum-instance.mdx [hunk 3]
```diff
@@ -12,21 +12,30 @@ These include measurement error mitigation, splitting and combining execution to
...
+| [`qiskit.utils.QuantumInstance.assemble`](/api/qiskit/0.46/qiskit.utils.QuantumInstance#assemble) | [`qiskit.compiler.assemble`](/api/qiskit/compiler#qiskit.compiler.assemble) [DEPRECATED: no longer part of the workflow]|
 
 The remainder of this guide focused on the [`qiskit.utils.QuantumInstance.execute`](/api/qiskit/0.46/qiskit.utils.QuantumInstance#execute) to
 [`qiskit.primitives`](/api/qiskit/primitives) migration path.
 
+<Admonition type="caution">
+  **Deprecation Notice**
+
+  This guide preceeds the introduction of Primitives V2. It was drafted with the intention to guide users from `QuantumInstance` to 
+  the Primitives V1 interface. Following the introduction of Primitives V2, some providers have deprecated Primitives V1
```
Comment (beckykd) [inline docs/migration-guides/qiskit-quantum-instance.mdx:34]:
```suggestion
  the V1 primitives interface. Following the introduction of the V2 primitives, some providers have deprecated V1 primitive 
```

**Review by beckykd — APPROVED** [review]

**Comment by beckykd** [comment]
I think this is a reasonable approach.

**Review by Eric-Arellano — APPROVED** [review]

**Review by javabster — APPROVED** [review]
LGTM 🚀

**Pull request merged**